// @azure/monitor-query-logs/dist/commonjs/models/public.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.KnownMonitorLogsQueryAudience = exports.LogsQueryResultStatus = void 0;
/** The status of a query. */
var LogsQueryResultStatus;
(function (LogsQueryResultStatus) {
    /** The query was successful. */
    LogsQueryResultStatus["Success"] = "Success";
    /** The query was partially successful. */
    LogsQueryResultStatus["PartialFailure"] = "PartialFailure";
    /** The query failed. */
    LogsQueryResultStatus["Failure"] = "Failure";
})(LogsQueryResultStatus || (exports.LogsQueryResultStatus = LogsQueryResultStatus = {}));
/** Audiences for Azure Monitor Logs. */
var KnownMonitorLogsQueryAudience;
(function (KnownMonitorLogsQueryAudience) {
    /** The Azure public cloud. */
    KnownMonitorLogsQueryAudience["AzurePublicCloud"] = "https://api.loganalytics.io";
    /** The Azure China cloud. */
    KnownMonitorLogsQueryAudience["AzureChina"] = "https://api.loganalytics.azure.cn";
    /** The Azure US Government cloud. */
    KnownMonitorLogsQueryAudience["AzureGovernment"] = "https://api.loganalytics.us";
})(KnownMonitorLogsQueryAudience || (exports.KnownMonitorLogsQueryAudience = KnownMonitorLogsQueryAudience = {}));
