// @azure/monitor-query-logs/dist/esm/monitorQueryLogsClient.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { createMonitorQueryLogs } from "./api/index.js";
import { batch, executeWithResourceId, execute } from "./api/operations.js";
export class LogsQueryClient {
    _client;
    /** The pipeline used by this client to make requests */
    pipeline;
    constructor(credential, options = {}) {
        const prefixFromOptions = options?.userAgentOptions?.userAgentPrefix;
        const userAgentPrefix = prefixFromOptions
            ? `${prefixFromOptions} azsdk-js-client`
            : `azsdk-js-client`;
        this._client = createMonitorQueryLogs(credential, {
            ...options,
            userAgentOptions: { userAgentPrefix },
        });
        this.pipeline = this._client.pipeline;
    }
    /**
     * Executes a batch of Analytics queries for data.
     * [Here](https://learn.microsoft.com/azure/azure-monitor/logs/api/batch-queries)
     * is an example for using POST with an Analytics query.
     */
    batch(body, options = { requestOptions: {} }) {
        return batch(this._client, body, options);
    }
    /**
     * Executes an Analytics query for data in the context of a resource.
     * [Here](https://learn.microsoft.com/azure/azure-monitor/logs/api/azure-resource-queries)
     * is an example for using POST with an Analytics query.
     */
    executeWithResourceId(resourceId, body, options = { requestOptions: {} }) {
        return executeWithResourceId(this._client, resourceId, body, options);
    }
    /**
     * Executes an Analytics query for data.
     * [Here](https://learn.microsoft.com/azure/azure-monitor/logs/api/request-format)
     * is an example for using POST with an Analytics query.
     */
    execute(workspaceId, body, options = { requestOptions: {} }) {
        return execute(this._client, workspaceId, body, options);
    }
}
