// @azure/monitor-query-logs/dist/esm/models/public.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
/** The status of a query. */
export var LogsQueryResultStatus;
(function (LogsQueryResultStatus) {
    /** The query was successful. */
    LogsQueryResultStatus["Success"] = "Success";
    /** The query was partially successful. */
    LogsQueryResultStatus["PartialFailure"] = "PartialFailure";
    /** The query failed. */
    LogsQueryResultStatus["Failure"] = "Failure";
})(LogsQueryResultStatus || (LogsQueryResultStatus = {}));
/** Audiences for Azure Monitor Logs. */
export var KnownMonitorLogsQueryAudience;
(function (KnownMonitorLogsQueryAudience) {
    /** The Azure public cloud. */
    KnownMonitorLogsQueryAudience["AzurePublicCloud"] = "https://api.loganalytics.io";
    /** The Azure China cloud. */
    KnownMonitorLogsQueryAudience["AzureChina"] = "https://api.loganalytics.azure.cn";
    /** The Azure US Government cloud. */
    KnownMonitorLogsQueryAudience["AzureGovernment"] = "https://api.loganalytics.us";
})(KnownMonitorLogsQueryAudience || (KnownMonitorLogsQueryAudience = {}));
